src/server.py

```diff
@@ -0,0 +1,45 @@
+"""functions to build the server side of echo."""
+
+
+from __future__ import unicode_literals
+import socket
+
+
+def server():
+    """Set up the server side to listen."""
+    server = socket.socket(
+        socket.AF_INET,
+        socket.SOCK_STREAM,
+        socket.IPPROTO_TCP
+    )
+    address = ('127.0.0.1', 5016)
+    server.bind(address)
+
+    server.listen(1)
+    print("Listening...")
+
+    conn, addr = server.accept()
+
+    while True:
+        try:
+            buffer_length = 8
+            response = u''
+            while response[-2:] != u"\r\n":
+                part = conn.recv(buffer_length)
+                response += part.decode('utf8')
```

Comment:
So you're decoding each chunk as you get it. Since unicode characters can take up multiple bytes, what happens when a multiple-byte unicode character isn't received all the way in one chunk? i.e. the buffer length cuts it in half?

Rather than decoding one chunk at a time, what would be a better approach?

And anyway, for the purpose of this echo server, do you really need to decode the message at the server end, only to re-encode it and send it back?